hbase-http/src/main/java/org/apache/hadoop/hbase/http/HttpServer.java

```diff
@@ -850,8 +853,13 @@ protected void addDefaultServlets(ContextHandlerCollection contexts, Configurati
         ServletConfig servletConfig = METRIC_SERVLETS.get(enabledServlet);
         if (servletConfig != null) {
           Class<?> clz = Class.forName(servletConfig.getClazz());
-          addPrivilegedServlet(servletConfig.getName(), servletConfig.getPathSpec(),
+          if (conf.getBoolean(HTTP_PRIVILEGED_METRICS_KEY, HTTP_PRIVILEGED_METRICS_DEFAULT)) {
+            addPrivilegedServlet(servletConfig.getName(), servletConfig.getPathSpec(),
             clz.asSubclass(HttpServlet.class));
+          } else {
+            addUnprivilegedServlet(servletConfig.getName(), servletConfig.getPathSpec(),
+            clz.asSubclass(HttpServlet.class));
```

Comment:
I want to remove kerberos authentication completely to collect metrics easily on kerberized hbase cluster using prometheus.
Even though using `.endpoint.whitelist` mechanism on the KerberosAuthenticationHandler, `addPrivilegedServlet()` still disallow unauthenticated users and also authenticated users but without admin privileges.
In order to support an option to disable kerberos authentication on metrics endpoints, both `.endpoint.whitelist` mechanism and `addUnprivilegedServlet()`  should be used.